src/irmin-pack/unix/ext.ml

```diff
@@ -555,7 +555,9 @@ module Maker (Config : Conf.S) = struct
           let on_finalise t f =
             match t.running_gc with
             | None -> ()
-            | Some x -> Lwt.on_success x.promise f
+            | Some x ->
+                let _ = Lwt.bind x.promise f in
```

Comment:
This is the main change

Reply:
I would expect this to also return a `unit Lwt.t` now. You could just return the result of the call to `Lwt.bind` - I'm not sure that it's safe to ignore promises like that.

Reply:
The reason it doesn't return `unit Lwt.t` is that the purpose of the `on_finalise` function is to allow callers to register asynchronous callbacks (as `Gc.run` does). So, we don't actually want to chain to the returned promise. This is safe because `bind` adds our callback to `t.running_gc.promise` which is kept around for the lifetime of a gc.